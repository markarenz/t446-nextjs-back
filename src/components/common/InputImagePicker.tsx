import { useState } from 'react';
import IconButton from './IconButton';
import { Asset } from '../../types/types';
import { IconClose } from '../img/icons';
import ImageUploader from './ImageUploader';
import ImageSelector from './ImageSelector';
import styles from '../../styles/modules/InputImagePicker.module.scss';

type Props = {
  name: string;
  value: string;
  onChange: Function;
  baseImgUrl: string;
  assetList: Asset[];
  handleTriggerRefresh: Function;
};
const InputImagePicker: React.FC<Props> = ({
  name,
  value,
  onChange,
  baseImgUrl,
  assetList,
  handleTriggerRefresh
}) => {
  const [isPickingImage, setIsPickingImage] = useState(false);
  const handleImageSelection = (filename: string) => {
    onChange(name, filename);
  };
  const handleCloseModal = () => {
    setIsPickingImage(false);
  };
  const handleOpenModal = () => {
    setIsPickingImage(true);
  };
  const handlePickItem = (filename: string) => {
    onChange(name, filename);
    handleCloseModal();
  };
  return (
    <div>
      <div
        className={styles.item}
        onClick={handleOpenModal}
        style={{
          backgroundImage: `url(${baseImgUrl}thumbnails/${
            value.length > 0 ? value : 'unknown.jpg'
          })`
        }}
      ></div>
      {isPickingImage && (
        <div className={styles.modalWrap}>
          <div className={styles.pickerModalWrap}>
            <IconButton title="close" onClick={handleCloseModal}>
              <IconClose />
            </IconButton>
            <div className={styles.inner}>
              <ImageUploader handleTriggerRefresh={handleTriggerRefresh} />
              <ImageSelector
                allowSelection={true}
                assetList={assetList}
                handleTriggerRefresh={handleTriggerRefresh}
                handleSelection={handlePickItem}
              />
            </div>
          </div>
        </div>
      )}
    </div>
  );
};
export default InputImagePicker;
