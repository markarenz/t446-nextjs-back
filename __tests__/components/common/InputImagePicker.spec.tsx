import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { getAssetList } from '../../__fixtures__/assetList';
import InputImagePicker from '../../../src/components/common/InputImagePicker';

const assetList = getAssetList();
const defaultProps = {
  name: 'image',
  value: 'test.jpg',
  onChange: jest.fn(),
  baseImgUrl: 'https://www.domain.com/',
  assetList,
  handleTriggerRefresh: jest.fn()
};
const altProps = {
  ...defaultProps,
  value: ''
};
describe('InputImagePicker', () => {
  it('renders imagePicker with value', async () => {
    render(<InputImagePicker {...defaultProps} />);
    const element = await screen.findByTestId('input-image-picker');
    expect(element).toBeInTheDocument();
  });
  it('renders imagePicker without value', async () => {
    render(<InputImagePicker {...altProps} />);
    const element = await screen.findByTestId('input-image-picker');
    expect(element).toBeInTheDocument();
  });
  it('opens modal on click', async () => {
    render(<InputImagePicker {...defaultProps} />);
    const element = await screen.findByTestId('input-image-picker-opener');
    fireEvent(
      element,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true
      })
    );
    const modal = await screen.findByTestId('image-picker-modal');
    expect(modal).toBeInTheDocument();
  });

  it('handles image selection', async () => {
    render(<InputImagePicker {...defaultProps} />);
    const element = await screen.findByTestId('input-image-picker-opener');
    fireEvent(
      element,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true
      })
    );
    const item = await screen.findByTestId('asset-item-0');
    fireEvent(
      item,
      new MouseEvent('click', {
        bubbles: true,
        cancelable: true
      })
    );
    expect(defaultProps.onChange).toHaveBeenCalled();
  });
});
