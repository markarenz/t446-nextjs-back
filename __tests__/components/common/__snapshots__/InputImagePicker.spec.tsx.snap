// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputImagePicker renders imagePicker with value 1`] = `
<div
  data-testid="input-image-picker"
>
  <div
    class="item"
    data-testid="input-image-picker-opener"
    style="background-image: url(https://www.domain.com/thumbnails/test.jpg);"
  />
</div>
`;
